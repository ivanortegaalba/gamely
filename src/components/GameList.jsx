import React from 'react';
import PropTypes from '../PropTypes';
import { FlexGrid, Grid } from '../style/styled-components/objects/Grid';
import Game from './Game';

// TODO: Create style-component List
export function GameList({
  games,
  open,
  addToFavourites,
  removeFromFavourites,
}) {
  const Wrapper = open ? Grid : FlexGrid;

  return (
    <Wrapper>
      {games.map((game, idx) => (
        <Game
          key={idx}
          game={game}
          addToFavourites={addToFavourites}
          removeFromFavourites={removeFromFavourites}
        />
      ))}
    </Wrapper>
  );
}

GameList.propTypes = {
  games: PropTypes.arrayOf(PropTypes.game),
};
GameList.defaultProps = {
  open: false,
  games: [],
};

export default GameList;
