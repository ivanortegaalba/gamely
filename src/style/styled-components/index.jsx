export { theme } from './theme'

// Components
export { Header } from './components/Header'
export { Button } from './components/Button'

// Elements
export {H1, H2, H3, H4, H5, H6} from './elements/Headings'

// Settings
export { colors } from './settings/colors'
export { fonts } from './settings/fonts'

// Tools
export { px2rems } from './tools/functions'
