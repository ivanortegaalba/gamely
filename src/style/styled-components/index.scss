@import '~sanitize.css';

// Settings
@import 'settings/colors';
@import 'settings/fonts';

// Tools
@import 'tools/functions';

// Components
@import 'components/button';
@import 'components/input';
@import 'components/badge';
@import 'components/table-list';
@import 'components/form-control';
@import 'components/header';

// Elements
@import 'elements/headings';

// Objects
@import 'objects/grid';
@import 'objects/spacing';
@import 'objects/flex';

body {
  font-family: $base-font-family;
  -webkit-font-smoothing: antialiased;
}
