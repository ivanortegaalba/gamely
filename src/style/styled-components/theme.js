import { colors } from './settings/colors';
import { fonts } from './settings/fonts';
import { spacing } from './settings/spacing';
import { fontSizes } from './settings/fontSizes';

export const theme = {
  fonts: { ...fonts },
  palette: { ...colors },
  spacing: { ...spacing },
  fontSizes: { ...fontSizes },
};
