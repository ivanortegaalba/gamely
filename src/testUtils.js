import 'jest-styled-components';
import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';

export const shallowWithTheme = (component, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();
  return shallow(component, { context });
};

export const mountWithTheme = (component, theme) => {
  const context = mount(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();
  return mount(component, { context });
};

export const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, { context });
};
