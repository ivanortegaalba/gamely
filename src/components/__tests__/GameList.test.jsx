import React from 'react';
import mockGame from '../../__mocks__/game';
import { shallowWithTheme } from '../../testUtils';
import { theme } from '../../style/styled-components/theme';
import { GameList } from '../GameList';

describe('<GameList/>', () => {
  describe('is empty', () => {
    const removeFromFavourites = jest.fn();
    const addToFavourites = jest.fn();
    const component = (
      <GameList
        removeFromFavourites={removeFromFavourites}
        addToFavourites={addToFavourites}
      />
    );
    const wrapper = shallowWithTheme(component, theme);
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('has many games', () => {
    const removeFromFavourites = jest.fn();
    const addToFavourites = jest.fn();
    const component = (
      <GameList
        games={[{ ...mockGame, isFavourite: true }, { ...mockGame }]}
        removeFromFavourites={removeFromFavourites}
        addToFavourites={addToFavourites}
      />
    );
    const wrapper = shallowWithTheme(component, theme);
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should render a Game component for each game in the list', () => {
      expect(wrapper.find('Game')).toHaveLength(2);
    });
    it('should pass correct props for each game', () => {
      expect(wrapper.find('Game')).toHaveLength(2);
    });
  });
});
