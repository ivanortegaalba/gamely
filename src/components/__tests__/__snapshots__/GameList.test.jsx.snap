// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameList/> has many games should render correctly 1`] = `
<styled.section>
  <Game
    addToFavourites={[Function]}
    game={
      Object {
        "hasBoosters": false,
        "isFavourite": true,
        "name": "8-Ball Pool",
        "short": "eightballpool",
        "tags": "",
        "url": "/spel/sportspel/8-ball-pool/?language=sv",
      }
    }
    key="0"
    removeFromFavourites={[Function]}
  />
  <Game
    addToFavourites={[Function]}
    game={
      Object {
        "hasBoosters": false,
        "name": "8-Ball Pool",
        "short": "eightballpool",
        "tags": "",
        "url": "/spel/sportspel/8-ball-pool/?language=sv",
      }
    }
    key="1"
    removeFromFavourites={[Function]}
  />
</styled.section>
`;

exports[`<GameList/> is empty should render correctly 1`] = `<styled.section />`;
